flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -219,52 +249,47 @@ public synchronized void stop() {
   @Override
   public Status process() throws EventDeliveryException {
     if (writer == null) {
-      try {
-        this.writer = newWriter(login, target);
-      } catch (DatasetException e) {
-        // DatasetException includes DatasetNotFoundException
-        throw new EventDeliveryException(
-          "Cannot write to " + getName(), e);
-      }
+      this.writer = newWriter();
     }
 
-    // handle file rolling
-    if ((System.currentTimeMillis() - lastRolledMs) / 1000 > rollIntervalS) {
-      // close the current writer and get a new one
-      writer.close();
-      this.writer = newWriter(login, target);
-      this.lastRolledMs = System.currentTimeMillis();
-      LOG.info("Rolled writer for " + getName());
+    if (shouldRoll()) {
```

Comment:
Ok, I refactored things a bit and I'll a null writer as a condition for rolling the writer since closeWriter() will check in the writer is null before trying to close it and it will only log when the writer is successfully closed.